migen/fhdl/specials.py

```diff
@@ -388,7 +388,7 @@ def gn(e):
             content = ""
             formatter = "{:0" + str(int(memory.width / 4)) + "X}\n"
             for d in memory.init:
-                content += formatter.format(d)
+                content += formatter.format(d).zfill(-(-memory.width // 4))
```

Comment:
```suggestion
                content += formatter.format(d).zfill(int(-(-memory.width // 4)))
```

Reply:
zfill expects an integer, not a float

Reply:
`//` is python's integer floor division operator...so `int()` shouldn't be necessary.

Reply:
That may be so, but in practice I was seeing syntax errors on zfill() because it was still spitting out 8.0 instead of 8.

Reply:
then imho the bug is wherever `memory.width` gets set to a float...

Reply:
It did that even with memory.width manually set to 32...

I just don't want to end up carrying a downstream patch to deal with specific Python version(s) that are behaving in this manner. :wink: